miri-script/src/commands.rs

```diff
@@ -172,7 +172,13 @@ impl Command {
             Command::Install { flags } => Self::install(flags),
             Command::Build { flags } => Self::build(flags),
             Command::Check { flags } => Self::check(flags),
-            Command::Test { bless, flags, target } => Self::test(bless, flags, target),
+            Command::Test { bless, flags, target, coverage } =>
+                Self::test(
+                    bless,
+                    flags,
+                    target,
+                    coverage.then_some(crate::coverage::CoverageReport::new()?),
```

Comment:
Please move this logic into the `test` function; the arguments should corresponds to the enum variant.